### PR TITLE
DAOS-10383 test: Improve failure reporting for enospace test.

### PR DESCRIPTION
**Pull request by ashleypittman**
- DAOS-623 engine: Update a warning in the IV code.
- DAOS-10383 ftest: Improve failure reporting for enospace test.

### CODE REVIEW

File: src/tests/ftest/dfuse/sparse_file.py [hunk 1]
```diff
@@ -77,15 +77,15 @@ def test_sparsefile(self):
         fsize_after_truncate = get_remote_file_size(self.hostlist_clients[0], sparse_file)
         self.log.info("File size (in bytes) after truncate: %s", fsize_after_truncate)
         # verifying the file size got set to desired value
-        self.assertTrue(fsize_after_truncate == self.space_before)
+        self.assertEqual(fsize_after_truncate, self.space_before)
```
Comment (phender) [inline src/tests/ftest/dfuse/sparse_file.py:80]: Could we add a descriptive message here that will be displayed if the assert fails?

File: src/tests/ftest/dfuse/sparse_file.py [hunk 2]
```diff
@@ -77,15 +77,15 @@ def test_sparsefile(self):
...
 
         # write to the first byte of the file with char 'A'
         dd_first_byte = "echo 'A' | dd conv=notrunc of={} bs=1 count=1".format(sparse_file)
         self.execute_cmd(dd_first_byte)
         fsize_write_1stbyte = get_remote_file_size(self.hostlist_clients[0], sparse_file)
         self.log.info("File size (in bytes) after writing first byte: %s", fsize_write_1stbyte)
         # verify file did not got overwritten after dd write.
-        self.assertTrue(fsize_write_1stbyte == self.space_before)
+        self.assertEqual(fsize_write_1stbyte, self.space_before)
```
Comment (phender) [inline src/tests/ftest/dfuse/sparse_file.py:88]: Could we add a descriptive message here that will be displayed if the assert fails?

File: src/tests/ftest/dfuse/sparse_file.py [hunk 3]
```diff
@@ -94,14 +94,14 @@ def test_sparsefile(self):
         fsize_write_1024thwrite = get_remote_file_size(self.hostlist_clients[0], sparse_file)
         self.log.info("File size (in bytes) after writing 1024th byte: %s", fsize_write_1024thwrite)
         # verify file did not got overwritten after dd write.
-        self.assertTrue(fsize_write_1024thwrite == self.space_before)
+        self.assertEqual(fsize_write_1024thwrite, self.space_before)
```
Comment (phender) [inline src/tests/ftest/dfuse/sparse_file.py:97]: Could we add a descriptive message here that will be displayed if the assert fails?

File: src/tests/ftest/dfuse/sparse_file.py [hunk 4]
```diff
@@ -94,14 +94,14 @@ def test_sparsefile(self):
...
 
         # Obtain the value of 1st byte and 1024th byte in the file and
         # compare their values, they should be same.
         check_first_byte = file_obj.read(1)
         file_obj.seek(1022, 1)
         check_1024th_byte = file_obj.read(1)
-        self.assertTrue(check_first_byte == check_1024th_byte)
+        self.assertEqual(check_first_byte, check_1024th_byte)
```
Comment (phender) [inline src/tests/ftest/dfuse/sparse_file.py:104]: Could we add a descriptive message here that will be displayed if the assert fails?

**Review by phender — CHANGES_REQUESTED** [review]

File: src/tests/ftest/dfuse/sparse_file.py (hunk 1 above)
Comment (ashleypittman) [reply src/tests/ftest/dfuse/sparse_file.py:80]: I see where you're coming from but I'm not sure what this test is actually doing.  I might copy the comment above in as a message.

Comment (daltonbohning) [reply src/tests/ftest/dfuse/sparse_file.py:80]: From a quick look, maybe
"File size incorrect after truncate"

File: src/tests/ftest/dfuse/sparse_file.py (hunk 3 above)
Comment (daltonbohning) [reply src/tests/ftest/dfuse/sparse_file.py:97]: From a quick look, maybe
"File size incorrect after write"

File: src/tests/ftest/dfuse/sparse_file.py (hunk 4 above)
Comment (daltonbohning) [reply src/tests/ftest/dfuse/sparse_file.py:104]: From a quick look, maybe
"Unexpected data in sparse file"

**Review by daltonbohning — APPROVED** [review]

**Comment by ashleypittman** [comment]
@daos-stack/daos-gatekeeper will need force-landing as it only ran the tests it modified.

**Pull request merged**